azurerm/internal/services/blueprints/blueprint_assignment_resource.go

```diff
@@ -0,0 +1,336 @@
+			if len(excludedPrincipalsRaw) != 0 {
+				assignmentLockSettings.ExcludedPrincipals = utils.ExpandStringSlice(excludedPrincipalsRaw)
+			}
+		}
+		assignment.AssignmentProperties.Locks = assignmentLockSettings
+	}
+
+	identity, err := expandArmBlueprintAssignmentIdentity(d.Get("identity").([]interface{}))
+	if err != nil {
+		return err
+	}
+	assignment.Identity = identity
+
+	if paramValuesRaw := d.Get("parameter_values"); paramValuesRaw != "" {
+		assignment.Parameters = expandArmBlueprintAssignmentParameters(paramValuesRaw.(string))
+	} else {
+		assignment.Parameters = expandArmBlueprintAssignmentParameters("{}")
+	}
+
+	if resourceGroupsRaw := d.Get("resource_groups"); resourceGroupsRaw != "" {
+		assignment.ResourceGroups = expandArmBlueprintAssignmentResourceGroups(resourceGroupsRaw.(string))
+	} else {
+		assignment.ResourceGroups = expandArmBlueprintAssignmentResourceGroups("{}")
+	}
+
+	resp, err := client.CreateOrUpdate(ctx, targetScope, name, assignment)
+	if err != nil {
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			string(blueprint.Waiting),
+			string(blueprint.Validating),
+			string(blueprint.Creating),
+			string(blueprint.Deploying),
+			string(blueprint.Locking),
+		},
+		Target:  []string{string(blueprint.Succeeded)},
+		Refresh: blueprintAssignmentCreateStateRefreshFunc(ctx, client, targetScope, name),
+		Timeout: d.Timeout(schema.TimeoutCreate),
+	}
+
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("failed waiting for Blueprint Assignment %q (Scope %q): %+v", name, targetScope, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmBlueprintAssignmentRead(d, meta)
+}
+
+func resourceArmBlueprintAssignmentRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Blueprints.AssignmentsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.AssignmentID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceScope := id.Scope
+	assignmentName := id.Name
+
+	resp, err := client.Get(ctx, resourceScope, assignmentName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] the Blueprint Assignment %q does not exist - removing from state", assignmentName)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Read failed for Blueprint Assignment (%q): %+v", assignmentName, err)
+	}
+
+	if resp.Name != nil {
+		d.Set("name", resp.Name)
+	}
+
+	if resp.Scope != nil {
+		d.Set("target_subscription_id", resp.Scope)
+	}
+
+	if resp.Location != nil {
+		d.Set("location", azure.NormalizeLocation(*resp.Location))
+	}
+
+	if resp.Identity != nil {
+		d.Set("identity", flattenArmBlueprintAssignmentIdentity(resp.Identity))
+	}
+
+	if resp.AssignmentProperties != nil {
+		if resp.AssignmentProperties.BlueprintID != nil {
+			d.Set("version_id", resp.AssignmentProperties.BlueprintID)
+		}
+
+		if resp.Parameters != nil {
```

Comment:
It looks like this is outside of the scope of `resp.AssignmentProperties`. Thoughts on moving this out of the current if block?

Reply:
This is actually inside the scope, I'd used the "shorthand" path to the object.  I'll update for the full path as `resp.AssignmentProperties.Parameters` for clarity.